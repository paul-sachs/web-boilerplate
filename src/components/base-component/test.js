import test from 'ava';
import { shallow } from 'enzyme';
import React from 'react';
import BaseComponent from './';

test('foo6', t => {
	const exportDocument = shallow(<BaseComponent />);

	t.snapshot(exportDocument.node);
});
