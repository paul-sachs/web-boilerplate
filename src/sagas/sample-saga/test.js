import { expectSaga } from 'redux-saga-test-plan';
import test from 'ava';
import { delay } from 'redux-saga';
import { userClickedButton, systemSetAlert } from 'store/actions';
import saga from './';

test(async t => {
	// This is what we consider an integration test for a saga.
	// It describes what the saga is expected to perform.
	// In this case, we expect delay to be called,
	// and a new action to be placed onto the redux store, setting the alert.
	// Detailed documentation can be found here: http://redux-saga-test-plan.jeremyfairbank.com/integration-testing/
	// Please note that this is just an example. We prefer not to use delay because it will actually wait 1000 ms
	// during the test, which is suboptimal.
	await expectSaga(saga)
		.call(delay, 1000)
		.put.actionType(systemSetAlert.getType())
		.dispatch(userClickedButton())
		.run({ timeout: 1200, silenceTimeout: true });
	// Since this saga is a takeLatest saga, it's expected to go on forever. Need to silence the timeout.
	t.pass();
});
