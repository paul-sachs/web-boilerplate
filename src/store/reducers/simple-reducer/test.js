import test from 'ava';
import { increment } from 'store/actions';
import reducer from './';

test('foo6', t => {
	t.is(2, reducer(1, increment()));
});
