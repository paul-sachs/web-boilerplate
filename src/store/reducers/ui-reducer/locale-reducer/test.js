import test from 'ava';
import { systemSetLocale } from 'store/actions';

import reducer from './';

test(t => {
	const initialState = {
		localeCode: 'fr',
		messages: { old: 'stuff' }
	};
	const newState = reducer(initialState, systemSetLocale({ localeCode: 'en' }));

	t.is(newState.localeCode, 'en', 'Can set locale to english');
	t.notDeepEqual(newState.messages, initialState.messages, 'Changing locales changes messages.');
});

test(t => {
	const initialState = {
		localeCode: 'fr',
		messages: { old: 'stuff' }
	};
	const newState = reducer(initialState, systemSetLocale({ localeCode: 'something-else' }));

	t.is(newState.localeCode, 'fr', 'Setting invalid locale does not set the locale');
	t.is(newState.messages, initialState.messages, 'Setting invalid locale does not change messages');
});
