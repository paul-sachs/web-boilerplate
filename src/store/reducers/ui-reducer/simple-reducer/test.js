import test from 'ava';
import { userClickedButton } from 'store/actions';

import reducer from './';

test(t => {
	const initialState = {
		count: 99,
		inputText: 'yo'
	};
	const newState = reducer(initialState, userClickedButton());

	t.is(newState.count, 100, 'Successfully added 1');
});
