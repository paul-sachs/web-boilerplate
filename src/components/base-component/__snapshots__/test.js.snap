// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`foo6 1`] = `<div />`;
